File: data-r2dbc-repositories/src/main/java/com/example/demo/PostRepository.java
package com.example.demo;

import org.springframework.data.r2dbc.repository.R2dbcRepository;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

import java.util.UUID;

public interface PostRepository extends R2dbcRepository<Post, UUID> {
    public Flux<Post> findByTitleContains(String name);

    public Mono<Long> countByTitleContains(String name);

}

File: jooq-kotlin-co-gradle/src/main/kotlin/com/example/demo/PostRepository.kt
package com.example.demo

import kotlinx.coroutines.flow.Flow
import org.springframework.data.repository.kotlin.CoroutineCrudRepository
import java.util.*

interface PostRepository : CoroutineCrudRepository<Post, UUID>, PostRepositoryCustom {
    fun findByStatus(status: Status): Flow<Post>
}